modules/ballerina-core/src/main/java/org/ballerinalang/model/values/BMessage.java

```diff
@@ -249,6 +255,17 @@ public BMessage clone() {
                 this.value.getMessageDataSource() instanceof BallerinaMessageDataSource) {
             clonedMessage.setMessageDataSource(((BallerinaMessageDataSource) this.value.
                     getMessageDataSource()).clone());
+            isDataAdded = true;
+        }
+
+        // This code block is added to fix the bug related to workers when accessing message variable within
+        // resource. Fix #2623
+        if (!isDataAdded) {
+            try {
+                Thread.sleep(1);
```

Comment:
@sameera-jayasoma This is fixed with https://github.com/ballerinalang/ballerina/pull/2736/commits/afd7fea0f9523d24a94bd4f5fc0e59937bcf9fcc